@ehsavoie Can you please double check CorruptedLogTestCase#testCorruptedJournalNotSaved test and behaviour related to setting the value of property to "0"? I see this exception intermittently on Windows machines (25% by ssh connection to lab, 100% on QE CI):
```
java.lang.AssertionError: Couldn't find C:\j\asts-integration\becd73a0\testsuite\integration\manualmode\target\wildfly\standalone\data\activemq\journal\attic
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.assertTrue(Assert.java:42)
	at org.jboss.as.test.manualmode.messaging.CorruptedLogTestCase.testCorruptedJournalNotSaved(CorruptedLogTestCase.java:216)
```

It seems that attic folder is not created sometimes. Is related assert really needed in this scenario?

--- a/messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/MessagingSubsystemParser_13_0.java
+++ b/messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/MessagingSubsystemParser_13_0.java
@@ -195,6 +195,7 @@ public class MessagingSubsystemParser_13_0 extends PersistentResourceXMLParser {
                                         ServerDefinition.JOURNAL_COMPACT_PERCENTAGE,
                                         ServerDefinition.JOURNAL_COMPACT_MIN_FILES,
                                         ServerDefinition.JOURNAL_MAX_IO,
+                                        ServerDefinition.JOURNAL_MAX_ATTIC_FILES,
                                         ServerDefinition.CREATE_BINDINGS_DIR,
                                         ServerDefinition.CREATE_JOURNAL_DIR,
                                         ServerDefinition.JOURNAL_DATASOURCE,

--- a/messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/MessagingTransformerRegistration.java
+++ b/messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/MessagingTransformerRegistration.java
@@ -93,6 +93,7 @@ public class MessagingTransformerRegistration implements ExtensionTransformerReg
 
     private static void registerTransformers_WF_23(ResourceTransformationDescriptionBuilder subsystem) {
         ResourceTransformationDescriptionBuilder server = subsystem.addChildResource(MessagingExtension.SERVER_PATH);
+
         rejectDefinedAttributeWithDefaultValue(server,
                 ServerDefinition.NETWORK_CHECK_LIST,
                 ServerDefinition.NETWORK_CHECK_NIC,
@@ -104,7 +105,8 @@ public class MessagingTransformerRegistration implements ExtensionTransformerReg
                 ServerDefinition.CRITICAL_ANALYZER_CHECK_PERIOD,
                 ServerDefinition.CRITICAL_ANALYZER_ENABLED,
                 ServerDefinition.CRITICAL_ANALYZER_POLICY,
-                ServerDefinition.CRITICAL_ANALYZER_TIMEOUT);
+                ServerDefinition.CRITICAL_ANALYZER_TIMEOUT,
+                ServerDefinition.JOURNAL_MAX_ATTIC_FILES);
         server.discardOperations(PrintDataOperation.OPERATION_NAME);
     }
 

--- a/messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/ServerAdd.java
+++ b/messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/ServerAdd.java
@@ -73,6 +73,7 @@ import static org.wildfly.extension.messaging.activemq.ServerDefinition.JOURNAL_
 import static org.wildfly.extension.messaging.activemq.ServerDefinition.JOURNAL_JDBC_LOCK_RENEW_PERIOD;
 import static org.wildfly.extension.messaging.activemq.ServerDefinition.JOURNAL_JDBC_NETWORK_TIMEOUT;
 import static org.wildfly.extension.messaging.activemq.ServerDefinition.JOURNAL_LARGE_MESSAGES_TABLE;
+import static org.wildfly.extension.messaging.activemq.ServerDefinition.JOURNAL_MAX_ATTIC_FILES;
 import static org.wildfly.extension.messaging.activemq.ServerDefinition.JOURNAL_MAX_IO;
 import static org.wildfly.extension.messaging.activemq.ServerDefinition.JOURNAL_MESSAGES_TABLE;
 import static org.wildfly.extension.messaging.activemq.ServerDefinition.JOURNAL_MIN_FILES;
@@ -626,6 +627,7 @@ class ServerAdd extends AbstractAddStepHandler {
             configuration.setJournalFileOpenTimeout(JOURNAL_FILE_OPEN_TIMEOUT.resolveModelAttribute(context, model).asInt());
             configuration.setJournalSyncNonTransactional(JOURNAL_SYNC_NON_TRANSACTIONAL.resolveModelAttribute(context, model).asBoolean());
             configuration.setJournalSyncTransactional(JOURNAL_SYNC_TRANSACTIONAL.resolveModelAttribute(context, model).asBoolean());
+            configuration.setJournalMaxAtticFiles(JOURNAL_MAX_ATTIC_FILES.resolveModelAttribute(context, model).asInt());
             configuration.setLogJournalWriteRate(LOG_JOURNAL_WRITE_RATE.resolveModelAttribute(context, model).asBoolean());
 
             configuration.setManagementAddress(SimpleString.toSimpleString(MANAGEMENT_ADDRESS.resolveModelAttribute(context, model).asString()));

--- a/messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/ServerDefinition.java
+++ b/messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/ServerDefinition.java
@@ -442,6 +442,15 @@ public class ServerDefinition extends PersistentResourceDefinition {
             .setValidator(new EnumValidator<>(JournalType.class, true, true))
             .setRestartAllServices()
             .build();
+    public static final SimpleAttributeDefinition JOURNAL_MAX_ATTIC_FILES = create("journal-max-attic-files", ModelType.INT)
+            .setAttributeGroup("journal")
+            .setXmlName("max-attic-files")
+            .setDefaultValue(new ModelNode(10))
+            .setRequired(false)
+            .setAllowExpression(true)
+            .setRestartAllServices()
+            .build();
+
     /**
      * @see ActiveMQDefaultConfiguration#isDefaultJournalLogWriteRate
      */
@@ -911,7 +920,8 @@ public class ServerDefinition extends PersistentResourceDefinition {
             JOURNAL_MESSAGES_TABLE, JOURNAL_BINDINGS_TABLE, JOURNAL_JMS_BINDINGS_TABLE, JOURNAL_LARGE_MESSAGES_TABLE, JOURNAL_PAGE_STORE_TABLE,
             JOURNAL_NODE_MANAGER_STORE_TABLE,
             JOURNAL_SYNC_TRANSACTIONAL, JOURNAL_SYNC_NON_TRANSACTIONAL, LOG_JOURNAL_WRITE_RATE,
-            JOURNAL_FILE_SIZE, JOURNAL_MIN_FILES, JOURNAL_POOL_FILES, JOURNAL_FILE_OPEN_TIMEOUT, JOURNAL_COMPACT_PERCENTAGE, JOURNAL_COMPACT_MIN_FILES, JOURNAL_MAX_IO,
+            JOURNAL_FILE_SIZE, JOURNAL_MIN_FILES, JOURNAL_POOL_FILES, JOURNAL_FILE_OPEN_TIMEOUT, JOURNAL_COMPACT_PERCENTAGE,
+            JOURNAL_COMPACT_MIN_FILES, JOURNAL_MAX_IO, JOURNAL_MAX_ATTIC_FILES,
             PERF_BLAST_PAGES, RUN_SYNC_SPEED_TEST, SERVER_DUMP_INTERVAL, MEMORY_WARNING_THRESHOLD, MEMORY_MEASURE_INTERVAL,
             GLOBAL_MAX_DISK_USAGE, DISK_SCAN_PERIOD, GLOBAL_MAX_MEMORY_SIZE,
             NETWORK_CHECK_NIC, NETWORK_CHECK_PERIOD, NETWORK_CHECK_TIMEOUT,

--- a/messaging-activemq/src/main/resources/org/wildfly/extension/messaging/activemq/LocalDescriptions.properties
+++ b/messaging-activemq/src/main/resources/org/wildfly/extension/messaging/activemq/LocalDescriptions.properties
@@ -402,6 +402,7 @@ server.journal-jdbc-lock-renew-period=The renewal period for the HA lock to keep
 server.journal-jdbc-network-timeout=The timeout used by the JDBC connection to detect network issues.
 server.journal-large-messages-table=Name of the JDBC table to store the large messages.
 server.journal-max-io=The maximum number of write requests that can be in the AIO queue at any one time.
+server.journal-max-attic-files=The maximum number of journal files stored in the attic in case of corrupted journal files.
 server.journal-messages-table=Name of the JDBC table to store the messages.
 server.journal-min-files=How many journal files to pre-create.
 server.journal-node-manager-store-table=Name of the JDBC table to store the node manager.

--- a/messaging-activemq/src/main/resources/schema/wildfly-messaging-activemq_13_0.xsd
+++ b/messaging-activemq/src/main/resources/schema/wildfly-messaging-activemq_13_0.xsd
@@ -255,6 +255,7 @@
                     <xs:attribute name="global-max-memory-size" type="xs:long" />
                     <xs:attribute name="global-max-disk-usage" type="xs:int" />
                     <xs:attribute name="disk-scan-period" type="xs:int" />
+                    <xs:attribute name="max-attic-files" type="xs:int" default="10"/>
                 </xs:complexType>
             </xs:element>
 

--- a/messaging-activemq/src/test/java/org/wildfly/extension/messaging/activemq/MessagingActiveMQSubsystem_13_0_TestCase.java
+++ b/messaging-activemq/src/test/java/org/wildfly/extension/messaging/activemq/MessagingActiveMQSubsystem_13_0_TestCase.java
@@ -284,7 +284,8 @@ public class MessagingActiveMQSubsystem_13_0_TestCase extends AbstractSubsystemB
                                 ServerDefinition.CRITICAL_ANALYZER_ENABLED,
                                 ServerDefinition.CRITICAL_ANALYZER_CHECK_PERIOD,
                                 ServerDefinition.CRITICAL_ANALYZER_POLICY,
-                                ServerDefinition.CRITICAL_ANALYZER_TIMEOUT
+                                ServerDefinition.CRITICAL_ANALYZER_TIMEOUT,
+                                ServerDefinition.JOURNAL_MAX_ATTIC_FILES
                         ))
                 .addFailedAttribute(subsystemAddress.append(SERVER_PATH, REPLICATION_MASTER_PATH),
                         new ChangeToTrueConfig(HAAttributes.CHECK_FOR_LIVE_SERVER.getName()))
@@ -347,7 +348,8 @@ public class MessagingActiveMQSubsystem_13_0_TestCase extends AbstractSubsystemB
                                 ServerDefinition.CRITICAL_ANALYZER_ENABLED,
                                 ServerDefinition.CRITICAL_ANALYZER_CHECK_PERIOD,
                                 ServerDefinition.CRITICAL_ANALYZER_POLICY,
-                                ServerDefinition.CRITICAL_ANALYZER_TIMEOUT
+                                ServerDefinition.CRITICAL_ANALYZER_TIMEOUT,
+                                ServerDefinition.JOURNAL_MAX_ATTIC_FILES
                         ))
                 .addFailedAttribute(subsystemAddress.append(SERVER_PATH, POOLED_CONNECTION_FACTORY_PATH),
                         new FailedOperationTransformationConfig.NewAttributesConfig(ConnectionFactoryAttributes.Common.USE_TOPOLOGY))
@@ -373,7 +375,8 @@ public class MessagingActiveMQSubsystem_13_0_TestCase extends AbstractSubsystemB
                             ServerDefinition.CRITICAL_ANALYZER_ENABLED,
                             ServerDefinition.CRITICAL_ANALYZER_CHECK_PERIOD,
                             ServerDefinition.CRITICAL_ANALYZER_POLICY,
-                            ServerDefinition.CRITICAL_ANALYZER_TIMEOUT));
+                            ServerDefinition.CRITICAL_ANALYZER_TIMEOUT,
+                            ServerDefinition.JOURNAL_MAX_ATTIC_FILES));
             config.addFailedAttribute(subsystemAddress.append(SERVER_PATH, CONNECTION_FACTORY_PATH), new FailedOperationTransformationConfig.NewAttributesConfig(ConnectionFactoryAttributes.Common.USE_TOPOLOGY));
             config.addFailedAttribute(subsystemAddress.append(SERVER_PATH, POOLED_CONNECTION_FACTORY_PATH), new FailedOperationTransformationConfig.NewAttributesConfig(ConnectionFactoryAttributes.Common.USE_TOPOLOGY));
         } else if (messagingVersion.compareTo(MessagingExtension.VERSION_6_0_0) > 0 ) {
@@ -389,7 +392,8 @@ public class MessagingActiveMQSubsystem_13_0_TestCase extends AbstractSubsystemB
                     ServerDefinition.CRITICAL_ANALYZER_ENABLED,
                     ServerDefinition.CRITICAL_ANALYZER_CHECK_PERIOD,
                     ServerDefinition.CRITICAL_ANALYZER_POLICY,
-                    ServerDefinition.CRITICAL_ANALYZER_TIMEOUT
+                    ServerDefinition.CRITICAL_ANALYZER_TIMEOUT,
+                    ServerDefinition.JOURNAL_MAX_ATTIC_FILES
             ));
             config.addFailedAttribute(subsystemAddress.append(SERVER_PATH, CONNECTION_FACTORY_PATH), new FailedOperationTransformationConfig.NewAttributesConfig(ConnectionFactoryAttributes.Common.USE_TOPOLOGY));
             config.addFailedAttribute(subsystemAddress.append(SERVER_PATH, POOLED_CONNECTION_FACTORY_PATH), new FailedOperationTransformationConfig.NewAttributesConfig(ConnectionFactoryAttributes.Common.USE_TOPOLOGY));
@@ -505,7 +509,8 @@ public class MessagingActiveMQSubsystem_13_0_TestCase extends AbstractSubsystemB
                     ServerDefinition.CRITICAL_ANALYZER_ENABLED,
                     ServerDefinition.CRITICAL_ANALYZER_CHECK_PERIOD,
                     ServerDefinition.CRITICAL_ANALYZER_POLICY,
-                    ServerDefinition.CRITICAL_ANALYZER_TIMEOUT
+                    ServerDefinition.CRITICAL_ANALYZER_TIMEOUT,
+                    ServerDefinition.JOURNAL_MAX_ATTIC_FILES
             ));
         }
         ModelTestUtils.checkFailedTransformedBootOperations(mainServices, messagingVersion, ops, config);

--- a/messaging-activemq/src/test/resources/org/wildfly/extension/messaging/activemq/subsystem_13_0.xml
+++ b/messaging-activemq/src/test/resources/org/wildfly/extension/messaging/activemq/subsystem_13_0.xml
@@ -205,7 +205,8 @@
                 jms-bindings-table="${journal.jms.bindings.table:JMS_BINDINGS}"
                 large-messages-table="${journal.large-messages.table:LARGE_MESSAGES}"
                 page-store-table="${journal.page-store.table:PAGE_STORE}"
-                node-manager-store-table="${journal.node-manager-store-table:NODE_MANAGER_STORE}" />
+                node-manager-store-table="${journal.node-manager-store-table:NODE_MANAGER_STORE}"
+                max-attic-files="${journal.max-attic-files:15}"/>
 
         <statistics
             enabled="${statistics.enabled:true}"

--- a/messaging-activemq/src/test/resources/org/wildfly/extension/messaging/activemq/subsystem_13_0_reject_transform.xml
+++ b/messaging-activemq/src/test/resources/org/wildfly/extension/messaging/activemq/subsystem_13_0_reject_transform.xml
@@ -111,7 +111,8 @@
                  global-max-disk-usage="70"
                  global-max-memory-size="100000"
                  disk-scan-period="10000"
-                 file-open-timeout="7"/>
+                 file-open-timeout="7"
+                 max-attic-files="15"/>
 
        <network-isolation
             network-check-list="127.0.0.1,[::1]"

--- a/messaging-activemq/src/test/resources/org/wildfly/extension/messaging/activemq/subsystem_13_0_transform.xml
+++ b/messaging-activemq/src/test/resources/org/wildfly/extension/messaging/activemq/subsystem_13_0_transform.xml
@@ -44,7 +44,8 @@
                 create-journal-dir="${create.journal.dir:true}"
                 global-max-disk-usage="100"
                 disk-scan-period="5000"
-                global-max-memory-size="-1"/>
+                global-max-memory-size="-1"
+                max-attic-files="10"/>
 
         <statistics
                 enabled="${statistics.enabled:true}"

--- a/testsuite/integration/manualmode/src/test/java/org/jboss/as/test/manualmode/messaging/CorruptedLogTestCase.java
+++ b/testsuite/integration/manualmode/src/test/java/org/jboss/as/test/manualmode/messaging/CorruptedLogTestCase.java
@@ -1,0 +1,244 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2012, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.jboss.as.test.manualmode.messaging;
+
+import java.io.File;
+import java.io.IOException;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.as.controller.client.helpers.Operations;
+
+import java.net.UnknownHostException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.stream.Collectors;
+import java.util.Properties;
+import java.util.stream.Stream;
+import javax.jms.ConnectionFactory;
+import javax.jms.JMSConsumer;
+import javax.jms.JMSContext;
+import javax.jms.JMSProducer;
+import javax.jms.Queue;
+import javax.jms.TextMessage;
+import javax.naming.Context;
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+import org.apache.commons.lang3.RandomStringUtils;
+
+import org.jboss.arquillian.container.test.api.ContainerController;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.as.arquillian.container.ManagementClient;
+import org.jboss.as.repository.PathUtil;
+import org.jboss.as.test.integration.common.jms.JMSOperations;
+import org.jboss.as.test.integration.common.jms.JMSOperationsProvider;
+import org.jboss.as.test.shared.TestSuiteEnvironment;
+import org.jboss.as.test.shared.TimeoutUtil;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * Test to check that we can configure properly how many journal files are stored in the attic when artemis journal is corrupted.
+ * @author Emmanuel Hugonnet (c) 2020 Red Hat, Inc.
+ */
+@RunAsClient()
+@RunWith(Arquillian.class)
+public class CorruptedLogTestCase {
+
+    private static final String DEFAULT_FULL_JBOSSAS = "default-full-jbossas";
+
+    @ArquillianResource
+    protected static ContainerController container;
+
+    private ManagementClient managementClient;
+    private int counter = 0;
+
+    @Before
+    public void setup() throws Exception {
+        if (!container.isStarted(DEFAULT_FULL_JBOSSAS)) {
+            container.start(DEFAULT_FULL_JBOSSAS);
+        }
+        managementClient = createManagementClient();
+        JMSOperations jmsOperations = JMSOperationsProvider.getInstance(managementClient.getControllerClient());
+        setJournalSize(100L);
+        jmsOperations.createJmsQueue("corrupted", "java:jboss/exported/" + "queue/corrupted");
+        jmsOperations.close();
+        managementClient.close();
+        container.stop(DEFAULT_FULL_JBOSSAS);
+        PathUtil.deleteRecursively(getAtticPath());
+        container.start(DEFAULT_FULL_JBOSSAS);
+        managementClient = createManagementClient();
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        JMSOperations jmsOperations = JMSOperationsProvider.getInstance(managementClient.getControllerClient());
+        jmsOperations.removeJmsQueue("corrupted");
+        managementClient.getControllerClient().execute(Operations.createUndefineAttributeOperation(jmsOperations.getServerAddress(), "journal-file-size"));
+        managementClient.getControllerClient().execute(Operations.createUndefineAttributeOperation(jmsOperations.getServerAddress(), "journal-max-attic-files"));
+        jmsOperations.close();
+        managementClient.close();
+        if(counter > 1) {
+            container.stop(DEFAULT_FULL_JBOSSAS);
+        }
+        PathUtil.deleteRecursively(getAtticPath());
+    }
+
+    /**
+     * Set the journal file size to 100k.
+     * Send 100 messages of 20k.
+     * Set the journal file size to 200k.
+     * Restart the server.
+     * Consume the messages.
+     * Should have journal files in the attic.
+     *
+     * @throws javax.naming.NamingException
+     * @throws java.io.IOException
+     */
+    @Test
+    public void testCorruptedJournal() throws NamingException, IOException {
+        counter++;
+        InitialContext remoteContext = createJNDIContext();
+        ConnectionFactory cf = (ConnectionFactory) remoteContext.lookup("jms/RemoteConnectionFactory");
+        Queue queue = (Queue) remoteContext.lookup("queue/corrupted");
+        try (JMSContext context = cf.createContext("guest", "guest", JMSContext.AUTO_ACKNOWLEDGE);) {
+            JMSProducer producer = context.createProducer();
+            for (int i = 0; i < 100; i++) {
+                producer.send(queue, context.createTextMessage(RandomStringUtils.randomAlphabetic(20000)));
+            }
+        }
+        setJournalSize(200L);
+        remoteContext.close();
+        managementClient.close();
+        container.stop(DEFAULT_FULL_JBOSSAS);
+        container.start(DEFAULT_FULL_JBOSSAS);
+        managementClient = createManagementClient();
+        remoteContext = createJNDIContext();
+        cf = (ConnectionFactory) remoteContext.lookup("jms/RemoteConnectionFactory");
+        queue = (Queue) remoteContext.lookup("queue/corrupted");
+        try (JMSContext context = cf.createContext("guest", "guest", JMSContext.AUTO_ACKNOWLEDGE);) {
+            JMSConsumer consumer = context.createConsumer(queue);
+            for (int i = 0; i < 100; i++) {
+                TextMessage message = (TextMessage) consumer.receive(TimeoutUtil.adjust(500));
+                Assert.assertNotNull(message);
+            }
+        }
+        Path attic = getAtticPath();
+        Assert.assertTrue("Couldn't find " + attic.toAbsolutePath(), Files.exists(attic));
+        Assert.assertTrue(Files.isDirectory(attic));
+        try (Stream<Path> stream = Files.list(attic)){
+            long nbAtticFiles = stream.collect(Collectors.counting());
+            Assert.assertTrue(nbAtticFiles > 0L);
+        }
+    }
+
+    /**
+     * Set the journal max attic files to 0 (aka don't store corrupted journal files).
+     * Set the journal file size to 100k.
+     * Send 100 messages of 20k.
+     * Set the journal file size to 200k.
+     * Restart the server.
+     * Consume the messages.
+     * Shouldn't have journal files in the attic.
+     *
+     * @throws javax.naming.NamingException
+     * @throws java.io.IOException
+     */
+    @Test
+    public void testCorruptedJournalNotSaved() throws NamingException, IOException {
+        Path attic = getAtticPath();
+        counter++;
+        JMSOperations jmsOperations = JMSOperationsProvider.getInstance(managementClient.getControllerClient());
+        managementClient.getControllerClient().execute(Operations.createWriteAttributeOperation(jmsOperations.getServerAddress(), "journal-max-attic-files", 0));
+        jmsOperations.close();
+        managementClient.close();
+        container.stop(DEFAULT_FULL_JBOSSAS);
+        PathUtil.deleteRecursively(attic);
+        container.start(DEFAULT_FULL_JBOSSAS);
+        Assert.assertFalse("Couldn't find " + attic.toAbsolutePath(), Files.exists(attic));
+        managementClient = createManagementClient();
+        InitialContext remoteContext = createJNDIContext();
+        ConnectionFactory cf = (ConnectionFactory) remoteContext.lookup("jms/RemoteConnectionFactory");
+        Queue queue = (Queue) remoteContext.lookup("queue/corrupted");
+        try (JMSContext context = cf.createContext("guest", "guest", JMSContext.AUTO_ACKNOWLEDGE);) {
+            JMSProducer producer = context.createProducer();
+            for (int i = 0; i < 100; i++) {
+                producer.send(queue, context.createTextMessage(RandomStringUtils.randomAlphabetic(20000)));
+            }
+        }
+        setJournalSize(200L);
+        remoteContext.close();
+        managementClient.close();
+        container.stop(DEFAULT_FULL_JBOSSAS);
+        container.start(DEFAULT_FULL_JBOSSAS);
+        managementClient = createManagementClient();
+        remoteContext = createJNDIContext();
+        cf = (ConnectionFactory) remoteContext.lookup("jms/RemoteConnectionFactory");
+        queue = (Queue) remoteContext.lookup("queue/corrupted");
+        try (JMSContext context = cf.createContext("guest", "guest", JMSContext.AUTO_ACKNOWLEDGE);) {
+            JMSConsumer consumer = context.createConsumer(queue);
+            for (int i = 0; i < 100; i++) {
+                TextMessage message = (TextMessage) consumer.receive(TimeoutUtil.adjust(500));
+                Assert.assertNotNull(message);
+            }
+        }
+        if (Files.exists(attic)) {
+            Assert.assertTrue(Files.isDirectory(attic));
+            try ( Stream<Path> stream = Files.list(attic)) {
+                long nbAtticFiles = stream.collect(Collectors.counting());
+                Assert.assertEquals("We shouldn't have any file in the attic", 0L, nbAtticFiles);
+            }
+        }
+    }
+
+    private void setJournalSize(long size) throws IOException {
+        JMSOperations jmsOperations = JMSOperationsProvider.getInstance(managementClient.getControllerClient());
+        managementClient.getControllerClient().execute(Operations.createWriteAttributeOperation(jmsOperations.getServerAddress(), "journal-file-size", 1024L * size));
+        jmsOperations.close();
+    }
+
+    private static ManagementClient createManagementClient() throws UnknownHostException {
+        return new ManagementClient(
+                TestSuiteEnvironment.getModelControllerClient(),
+                TestSuiteEnvironment.formatPossibleIpv6Address(TestSuiteEnvironment.getServerAddress()),
+                TestSuiteEnvironment.getServerPort(),
+                "remote+http");
+    }
+
+    protected static InitialContext createJNDIContext() throws NamingException {
+        final Properties env = new Properties();
+        env.put(Context.INITIAL_CONTEXT_FACTORY, "org.jboss.naming.remote.client.InitialContextFactory");
+        String ipAdddress = TestSuiteEnvironment.formatPossibleIpv6Address(TestSuiteEnvironment.getServerAddress());
+        env.put(Context.PROVIDER_URL, System.getProperty(Context.PROVIDER_URL, "remote+http://" + ipAdddress + ":8080"));
+        env.put(Context.SECURITY_PRINCIPAL, "guest");
+        env.put(Context.SECURITY_CREDENTIALS, "guest");
+        return new InitialContext(env);
+    }
+
+    private Path getAtticPath() {
+        Path jbossHome = new File(System.getProperty("jboss.home", "jboss-as")).toPath();
+        return jbossHome.resolve("standalone").resolve("data").resolve("activemq").resolve("journal").resolve("attic");
+    }
+
+}

